Mathlib/RingTheory/Perfectoid/FontaineTheta.lean

```diff
@@ -182,3 +187,26 @@ theorem fontaineTheta_teichmuller (x : R♭) : fontaineTheta R p (teichmuller p
   · simp [SModEq, mk_pow_fontaineTheta]
 
 end WittVector
+
+variable [Fact ¬IsUnit (p : R)] [IsAdicComplete (span {(p : R)}) R]
+
+/-- If the Frobenius map is surjective on `R/pR`, then the Fontaine's θ map is surjective. -/
+theorem surjective_fontaineTheta (hF : Function.Surjective (frobenius (ModP R p) p)) :
+    Function.Surjective (fontaineTheta R p) := by
+  have : Ideal.map (fontaineTheta R p) (span {(p : 𝕎 R♭)}) = 𝔭 := by
+    simp [map_span]
+  have _ : IsHausdorff ((span {(p : 𝕎 R♭)}).map (fontaineTheta R p)) R := by
+    rw [this]
+    infer_instance
+  apply surjective_of_mk_map_comp_surjective (fontaineTheta R p) (I := span {(p : 𝕎 R♭)})
+  simp only [RingHom.coe_comp]
+  suffices h : Function.Surjective (Ideal.Quotient.mk 𝔭 ∘ fontaineTheta R p) by
+    convert h
```

Comment:
Hmm, then you might as well use `change`.

But do you need to rely on defeq here? Or can `simpa using h` also work?